core/math/transform_2d.cpp

```diff
@@ -144,6 +156,18 @@ void Transform2D::translate_local(const Vector2 &p_translation) {
 	columns[2] += basis_xform(p_translation);
 }
 
+void Transform2D::orthogonalize() {
```

Comment:
The problem with a method like this is that its very misleading. You cant call multiple times without incurring into precision loss. The size will mutate over time and deform the object.

Reply:
The same thing has already been done in Basis, but I can't think of any other way. Definitely there is an accuracy issue, but I assume that calling it multiple times seems to be a corner case. Should I describe the possible loss of accuracy in the documentation?

Reply:
Added note:
```
Note: When a Basis/Transform2D that is already orthogonalized uses this method, the equality of the values before and after is not guaranteed.
```

Reply:
Instead of adding notes, I added a check `is_orthogonal()` to Basis and added a check `Math::is_zero_approx(get_skew())` to Transform2D for making guarantee the equality.

Reply:
One possibility for the Transform2D version is to always make the Y axis orthogonal to the X axis and leave the X axis untouched, this should more or less avoid precision errors.

Reply:
I implemented what you said, but the behavior is no different from setting skew = 0, so I don't think there is much point in having it. Well, I think it is fine to have the function for consistency with 3D.

Reply:
for now
```cpp
Vector2 y = Vector2(-(columns[0].y), columns[0].x).normalized(); // Make a vector orthogonal to x.
if (y.dot(columns[1].normalized()) < 0) {
	y = -y; // Flip to be closer to current y.
}
columns[1] = y * columns[1].length();
```